djcelery_email/tasks.py

```diff
@@ -44,7 +44,10 @@ def send_emails(messages, backend_kwargs=None, **kwargs):
     messages = [email_to_dict(m) for m in messages]
 
     conn = get_connection(backend=settings.CELERY_EMAIL_BACKEND, **combined_kwargs)
-    conn.open()
+    try:
+        conn.open()
+    except Exception:
+        logger.exception("Cannot reach CELERY_EMAIL_BACKEND %s", settings.CELERY_EMAIL_BACKEND)
```

Comment:
## Sending an email
```
[2016-01-19 11:06:53,184: INFO/MainProcess] Received task: djcelery_email_send_multiple[fd971ef7-ed10-490b-bdff-f48a83be7981]
[2016-01-19 11:06:53,241: INFO/MainProcess] Task djcelery_email_send_multiple[fd971ef7-ed10-490b-bdff-f48a83be7981] succeeded in 0.0544832569999s: 1
```

## Sending an email when email back-end is disabled:
```
[2016-01-19 11:08:02,354: INFO/MainProcess] Received task: djcelery_email_send_multiple[c543c4f6-f521-4fe4-a64f-9b3a8f0b3c0d]
[2016-01-19 11:08:02,531: ERROR/MainProcess] Task djcelery_email_send_multiple[c543c4f6-f521-4fe4-a64f-9b3a8f0b3c0d] raised unexpected: error(111, 'Connection refused')
Traceback (most recent call last):
  File "/home/tom/env_new/local/lib/python2.7/site-packages/celery/app/trace.py", line 240, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/home/tom/env_new/local/lib/python2.7/site-packages/celery/app/trace.py", line 438, in __protected_call__
    return self.run(*args, **kwargs)
  File "/home/tom/env_new/local/lib/python2.7/site-packages/djcelery_email/tasks.py", line 47, in send_emails
    conn.open()
  File "/home/tom/env_new/local/lib/python2.7/site-packages/django/core/mail/backends/smtp.py", line 58, in open
    self.connection = connection_class(self.host, self.port, **connection_params)
  File "/usr/lib/python2.7/smtplib.py", line 256, in __init__
    (code, msg) = self.connect(host, port)
  File "/usr/lib/python2.7/smtplib.py", line 316, in connect
    self.sock = self._get_socket(host, port, self.timeout)
  File "/usr/lib/python2.7/smtplib.py", line 291, in _get_socket
    return socket.create_connection((host, port), timeout)
  File "/usr/lib/python2.7/socket.py", line 575, in create_connection
    raise err
error: [Errno 111] Connection refused
[2016-01-19 11:08:02,533: ERROR/MainProcess] Pool callback raised exception: ValueError('too many values to unpack',)
Traceback (most recent call last):
  File "/home/tom/env_new/local/lib/python2.7/site-packages/billiard/pool.py", line 1700, in safe_apply_callback
    fun(*args, **kwargs)
  File "/home/tom/env_new/local/lib/python2.7/site-packages/celery/worker/job.py", line 384, in on_success
    return self.on_failure(ret_value)
  File "/home/tom/env_new/local/lib/python2.7/site-packages/celery/worker/job.py", line 443, in on_failure
    self._log_error(exc_info, send_failed_event=send_failed_event)
  File "/home/tom/env_new/local/lib/python2.7/site-packages/celery/worker/job.py", line 514, in _log_error
    task.send_error_email(context, einfo.exception)
  File "/home/tom/env_new/local/lib/python2.7/site-packages/celery/app/task.py", line 886, in send_error_email
    self.ErrorMail(self, **kwargs).send(context, exc)
  File "/home/tom/env_new/local/lib/python2.7/site-packages/celery/utils/mail.py", line 190, in send
    fail_silently=fail_silently)
  File "/home/tom/env_new/local/lib/python2.7/site-packages/celery/app/base.py", line 416, in mail_admins
    to = [admin_email for _, admin_email in self.conf.ADMINS]
ValueError: too many values to unpack
```

## Wrapping the `conn.open()`
```
[2016-01-19 11:13:20,374: INFO/MainProcess] Received task: djcelery_email_send_multiple[122d8b32-641a-4d76-a404-77bc11be5e8e] eta:[2016-01-19 11:16:20.342427-05:00]
[2016-01-19 11:15:55,513: ERROR/Worker-2] djcelery_email_send_multiple[b959a58e-920a-4e49-bd69-23a6f5eb87e2]: Cannot reach CELERY_EMAIL_BACKEND django.core.mail.backends.smtp.EmailBackend
Traceback (most recent call last):
  File "/home/tom/env_new/local/lib/python2.7/site-packages/djcelery_email/tasks.py", line 48, in send_emails
    conn.open()
  File "/home/tom/env_new/local/lib/python2.7/site-packages/django/core/mail/backends/smtp.py", line 58, in open
    self.connection = connection_class(self.host, self.port, **connection_params)
  File "/usr/lib/python2.7/smtplib.py", line 256, in __init__
    (code, msg) = self.connect(host, port)
  File "/usr/lib/python2.7/smtplib.py", line 316, in connect
    self.sock = self._get_socket(host, port, self.timeout)
  File "/usr/lib/python2.7/smtplib.py", line 291, in _get_socket
    return socket.create_connection((host, port), timeout)
  File "/usr/lib/python2.7/socket.py", line 575, in create_connection
    raise err
error: [Errno 111] Connection refused
[2016-01-19 11:15:55,516: WARNING/Worker-2] djcelery_email_send_multiple[b959a58e-920a-4e49-bd69-23a6f5eb87e2]: Failed to send email message to [u'to@example.com'], retrying. (error(111, 'Connection refused'))
[2016-01-19 11:15:55,555: INFO/MainProcess] Received task: djcelery_email_send_multiple[b959a58e-920a-4e49-bd69-23a6f5eb87e2] eta:[2016-01-19 11:18:55.519083-05:00]
[2016-01-19 11:15:55,555: INFO/MainProcess] Task djcelery_email_send_multiple[b959a58e-920a-4e49-bd69-23a6f5eb87e2] succeeded in 0.0441079919999s: 0
[2016-01-19 11:16:20,519: ERROR/Worker-3] djcelery_email_send_multiple[122d8b32-641a-4d76-a404-77bc11be5e8e]: Cannot reach CELERY_EMAIL_BACKEND django.core.mail.backends.smtp.EmailBackend
Traceback (most recent call last):
  File "/home/tom/env_new/local/lib/python2.7/site-packages/djcelery_email/tasks.py", line 48, in send_emails
    conn.open()
  File "/home/tom/env_new/local/lib/python2.7/site-packages/django/core/mail/backends/smtp.py", line 58, in open
    self.connection = connection_class(self.host, self.port, **connection_params)
  File "/usr/lib/python2.7/smtplib.py", line 256, in __init__
    (code, msg) = self.connect(host, port)
  File "/usr/lib/python2.7/smtplib.py", line 316, in connect
    self.sock = self._get_socket(host, port, self.timeout)
  File "/usr/lib/python2.7/smtplib.py", line 291, in _get_socket
    return socket.create_connection((host, port), timeout)
  File "/usr/lib/python2.7/socket.py", line 575, in create_connection
    raise err
error: [Errno 111] Connection refused
[2016-01-19 11:16:20,521: WARNING/Worker-3] djcelery_email_send_multiple[122d8b32-641a-4d76-a404-77bc11be5e8e]: Failed to send email message to [u'to@example.com'], retrying. (error(111, 'Connection refused'))
[2016-01-19 11:16:20,557: INFO/MainProcess] Task djcelery_email_send_multiple[122d8b32-641a-4d76-a404-77bc11be5e8e] succeeded in 0.040510624s: 0
[2016-01-19 11:16:20,558: INFO/MainProcess] Received task: djcelery_email_send_multiple[122d8b32-641a-4d76-a404-77bc11be5e8e] eta:[2016-01-19 11:19:20.523559-05:00]
```

So the main process will log the exception, and then the worker will log the same exception. Only the worker will continue to retry.